solr/core/src/java/org/apache/solr/security/JWTVerificationkeyResolver.java

```diff
@@ -110,7 +112,13 @@ public Key resolveKey(JsonWebSignature jws, List<JsonWebStructure> nestingContex
       if (issuerConfig.usesHttpsJwk()) {
         keysSource = "[" + String.join(", ", issuerConfig.getJwksUrls()) + "]";
         for (HttpsJwks hjwks : issuerConfig.getHttpsJwks()) {
-          jsonWebKeys.addAll(hjwks.getJsonWebKeys());
+          try {
+            jsonWebKeys.addAll(hjwks.getJsonWebKeys());
+          } catch (SSLHandshakeException e) {
+            log.warn("Failed to talk SSL with {}, do you have the correct SSL certificate configured?", hjwks.getLocation());
+            throw new SolrException(SolrException.ErrorCode.SERVER_ERROR,
+                "SSL error when validating auth token. Make sure correct certificates are configured.", e);
```

Comment:
Generally not a fan of both logging and throwing, as that likely means that we log twice.

Reply:
Agree, now only throws exception